### PR TITLE
Released COVIDx V9 dataset

### PR DESCRIPTION
**Pull request by mayaliliya**
Released the COVIDx9 dataset that maintains the same test set as COVIDx8 but has additional normal, pneumonia, and COVID-19 images into the training dataset from BIMCV, RSNA, and Stong Brook public datasets.

### CODE REVIEW

File: docs/COVIDx.md [hunk 1]
```diff
@@ -35,30 +52,30 @@ The current COVIDx dataset is constructed by the following open source chest rad
...
 |  test |        200        |         178       |  378  |
 
 
-COVIDx V7A 
+COVIDx V9A 
 Chest radiography images distribution
 |  Type | Normal | Pneumonia | COVID-19 | Total |
 |:-----:|:------:|:---------:|:--------:|:-----:|
-| train |  7966  |    5475   |   2158   | 15599 |
+| train |  8085  |    5555   |   16490  | 30130 |
```
Comment (haydengunraj) [inline docs/COVIDx.md:74]: What's the reason for the 352-image difference between 9A and 9B?

File: docs/COVIDx.md [hunk 2]
```diff
@@ -35,30 +52,30 @@ The current COVIDx dataset is constructed by the following open source chest rad
...
 |:-----:|:-----------------:|:-----------------:|:-----:|
-| train |       13794       |        2158       | 15952 |
+| train |       13992       |        16490      | 30482 |
 |  test |        200        |        200        |  400  |
 
 Patients distribution
 |  Type | COVID-19 Negative | COVID-19 Positive | Total |
 |:-----:|:-----------------:|:-----------------:|:-----:|
-| train |       13651       |        1327       | 14978 |
+| train |       13850       |        2808       | 16658 |
```
Comment (haydengunraj) [inline docs/COVIDx.md:66]: Do we have overlap between positive/negative patients? From my Kaggle download, I get the same class-wise counts but a different overall count (i.e., if I add the two class-wise counts it's the same as you have here, but if I disregard classes and just count the PIDs I get 10 fewer patients).

The commands I'm using are:
```sh
cat train.txt | grep " negative " | cut -f1 -d " " | sort | uniq | wc -l  # gives 13850 negative patients
cat train.txt | grep " positive " | cut -f1 -d " " | sort | uniq | wc -l  # gives 2808 positive patients
cat train.txt | cut -f1 -d " " | sort | uniq | wc -l                      # gives 16648 total patients instead of 16658
```

After checking the class-wise PIDs, it seems we do have overlap, although I'm not sure whether this is concerning or not. To get the list of overlapping PIDs, you can run

```sh
cat train.txt | grep " positive " | cut -f1 -d " " | sort | uniq > pos_pids.txt
cat train.txt | grep " negative " | cut -f1 -d " " | sort | uniq > neg_pids.txt
comm -12 pos_pids.txt neg_pids.txt
```
which gives this list:
```
173
313b
315
316
318
325
351
ANON58
ANON60
ANON96
```
A quick check of a few of these confirms that the same PIDs seem to be represented in both positive and negative cases:
```sh
cat train.txt | grep 313b | head
# >>> 313b 10.1016-slash-j.hrcr.2020.04.015-fig3-a.png negative cohen
# >>> 313b 10.1016-slash-j.hrcr.2020.04.015-fig3-b.png positive cohen

cat train.txt | grep ANON58
# >>> ANON58 CR.1.2.840.113564.1722810170.20200401171818781600.1003000225002.png negative actmed
# >>> ANON58 CR.1.2.840.113564.1722810170.20200326010813390400.1003000225002.png positive actmed

cat train.txt | grep ANON60
# >>> ANON60 DX.1.2.840.113564.1722810162.20200403105318208480.1203801020003.png negative actmed
# >>> ANON60 DX.1.2.840.113564.1722810162.20200408170334886370.1203801020003.png negative actmed
# >>> ANON60 DX.1.2.840.113564.1722810162.20200326123403551130.1203801020003.png positive actmed
# >>> ANON60 DX.1.2.840.113564.1722810162.20200326123552855170.1203801020003.png positive actmed
```

**Comment by haydengunraj** [comment]
A couple of minor points with respect to image and patient counts, but otherwise this looks pretty straightforward. I'm good to approve this once the concerns over counts are addressed.

File: docs/COVIDx.md (hunk 1 above)
Comment (mayaliliya) [reply docs/COVIDx.md:74]: When we transitioned to a binary dataset there were a couple of image from the previous datasets that we were uncomfortable including into "normal" and "pneumonia" bins, however, since we now have a more general negative bin they can be included.
This difference has been consistent since Covidx7

File: docs/COVIDx.md (hunk 2 above)
Comment (mayaliliya) [reply docs/COVIDx.md:66]: Thank you for the thorough investigation!

The overlap in patients was noted in COVIDx8 and was investigated to be found sound when we did the COVIDNet CXR-2 paper (this overlap count in patients is published in the paper).

The ANON60 patient from actmed has two positive COVID-19 CXRs and the 2 negative are labelled as no-finding
The ANON58 has a CXR that is COVID-19 positive and one that is no-finding
For 313b - on the choen metadata.csv the patientId is also the same for both images, with one as negative as a result of an unclear no-finding of COVID-19, and the fig3-b.png showcasing a positive COVID-19 detection (if you read the description explicitly). On visual inspection the CXR appear to be from the same patient (in terms of no visible extreme difference that would suggest a different anatomy)

**Comment by haydengunraj** [comment]
Thanks for the replies, I'm approving this now. The only thing to maybe consider is dropping the patient count down to 16648, as strictly speaking that's our actual total patient count.

**Review by haydengunraj — APPROVED** [review]

**Comment by mayaliliya** [comment]
> Thanks for the replies, I'm approving this now. The only thing to maybe consider is dropping the patient count down to 16648, as strictly speaking that's our actual total patient count.

Sounds good, I dropped it back down to 16,648. If there are questions regarding it moving forward we can more explicitly document the discrepancy as for now I will simply drop the total by 10.
